src/service/position-observer-impl.js

```diff
@@ -39,6 +44,33 @@ export const SEND_POSITIONS_HIGH_FIDELITY = 'send-positions-high-fidelity';
 export const POSITION_HIGH_FIDELITY = 'position-high-fidelity';
 
 /**
+* RelativeLocations
+*
+* Describes the relative position of an element to its viewport (whether the
+* element is inside the viewport, on top of the viewport or on the bottom
+* @enum {number}
+*/
+export const RelativeLocations = {
+  INSIDE: RelativePositions.INSIDE,
+  TOP: RelativePositions.TOP,
+  BOTTOM: RelativePositions.BOTTOM,
+};
+
+/**
+* VisibilityStates
+*
+* States used to describe whether the element is fully visible inside the
+* viewport, partially visible or out of view
+*
+* @enum {number}
+*/
+export const VisibilityStates = {
```

Comment:
Do we really want the calculation here? 
`RelativeLocations` is needed because if element is outside viewport, there's no way for them to figure out if they're above/below viewport. But `VisibilityStates` they can easily calculate from the `positionObserver` returned value.